code/datums/mood.dm

```diff
@@ -209,18 +212,23 @@
 /datum/mood/proc/update_mood()
 	if(QDELETED(mob_parent)) //don't bother updating their mood if they're about to be salty anyway. (in other words, we're about to be destroyed too anyway.)
 		return
+	var/something_bad_happened = FALSE
 	mood = 0
 	shown_mood = 0
 
 	SEND_SIGNAL(mob_parent, COMSIG_CARBON_MOOD_UPDATE)
 
 	for(var/category in mood_events)
 		var/datum/mood_event/the_event = mood_events[category]
+		var/mood_change = the_event.mood_change
+		if(mood_change > 0)
+			mood_change *= mood_modifier[MOOD_MODIFIER_GOOD]
+		else
+			something_bad_happened = TRUE
+			mood_change *= mood_modifier[MOOD_MODIFIER_BAD]
 		mood += the_event.mood_change
```

Comment:
I think this is supposed to use your new var and not the event's value, right?

Reply:
whoops, good catch